tactics/equality.ml

```diff
@@ -1075,24 +1075,20 @@ let onEquality with_evars tac (c,lbindc) =
   let state = Proofview.Goal.state gl in
   let t = Retyping.get_type_of env sigma c in
   let t' = try snd (Tacred.reduce_to_quantified_ind env sigma t) with UserError _ -> t in
-  let sigma, eq_clause = EClause.make_evar_clause env sigma t' in
-  let sigma = EClause.solve_evar_clause env sigma false eq_clause lbindc in
-  if not with_evars && List.exists (fun h -> h.EClause.hole_deps) eq_clause.EClause.cl_holes then
-    let filter h = if h.EClause.hole_deps then Some h.EClause.hole_name else None in
-    let bindings = List.map_filter filter eq_clause.EClause.cl_holes in
-    Proofview.tclZERO  (RefinerError (env, sigma, UnresolvedBindings bindings))
-  else
+  let sigma', eq_clause = EClause.make_evar_clause env sigma t' in
```

Comment:
Nitpick: I'm not a huge fan of having several names bound at toplevel for state passing values, it's easy to break code that way. If you desire so, you could wrap the building + check into a let binding to make the name more local, as:
```ocaml
let sigma, eq_clause =
  let sigma', eq_clause = EClause.make_evar_clause env sigma t' in
  let sigma' = EClause.solve_evar_clause env sigma' false eq_clause lbindc in
  let () = if not with_evars then EClause.check_evar_clause env sigma sigma' eq_clause in
  sigma', eq_clause
in
```

Reply:
Adopted!